### PR TITLE
docs: document qt.style and programs.rofi.theme changes

### PR DESCRIPTION
**Pull request by thiagokokada**
### Description

Add documentation for `qt.style` and `programs.rofi.theme` changes in release notes for 21.05 release.

### Checklist

- [x] Change is backwards compatible.

- [ ] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: doc/release-notes/rl-2105.adoc [hunk 1]
```diff
@@ -50,8 +50,8 @@ As a result of this change, <<opt-programs.mpv.package>> is no longer the
 resulting derivation. Use the newly introduced `programs.mpv.finalPackage`
 instead.
 
-* The <<opt-programs.rofi.extraConfig>> option is now an attrset rather
-than a string. To migrate, move the each line into the attrset,
+* The <<opt-programs.rofi.extraConfig>> option is now an attribute set rather
+than a string. To migrate, move the each line into the attribute set,
```
Comment (berbiche) [inline doc/release-notes/rl-2105.adoc:54]:
```suggestion
than a string. To migrate, move each line into the attribute set,
```

The `rofi` prefix needs to be dropped too, right?

File: doc/release-notes/rl-2105.adoc [hunk 2]
```diff
@@ -113,6 +141,21 @@ configure.customRC -> programs.neovim.extraConfig
 
 * Home Manager now respects the `NO_COLOR` environment variable as per
 https://no-color.org/[].
++
```
Comment (berbiche) [inline doc/release-notes/rl-2105.adoc:144]:
```suggestion

```

Remove the `+` here

File: doc/release-notes/rl-2105.adoc [hunk 3]
```diff
@@ -113,6 +141,21 @@ configure.customRC -> programs.neovim.extraConfig
...
+* Qt module now supports <<opt-qt.style.name>> to specify a theme name and <<opt-qt.style.package>>
+to specify a theme package. In case you're using <<opt-qt.platformTheme>> set to `gnome`, now it
+is required to set a theme that is compatible with both Qt and Gtk. For example:
```
Comment (berbiche) [inline doc/release-notes/rl-2105.adoc:147]:
```suggestion
to specify a theme package. If you have set <<opt-qt.platformTheme>> to `gnome`, a <<opt.qt.style.package>> compatible with both Qt and Gtk is now required to be set. For instance:
```

I'm not fond of either phrasing, feel free to improve.

File: doc/release-notes/rl-2105.adoc [hunk 4]
```diff
@@ -71,6 +71,34 @@ programs.rofi.extraConfig = {
   modi = "drun,emoji,ssh";
 };
 ----
++
+Also, <<opt-programs.rofi.theme>> option now supports defining a theme
+using an attribute set, so the following configuration is possible:
++
```
Comment (berbiche) [inline doc/release-notes/rl-2105.adoc:77]:
```suggestion
* The <<opt-programs.rofi.theme>> option now supports defining a theme
using an attribute set, the following configuration is now possible:
+
```

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for improving the doc.
Feel free to improve my suggestions.

File: doc/release-notes/rl-2105.adoc (hunk 1 above)
Comment (thiagokokada) [reply doc/release-notes/rl-2105.adoc:54]: Yeah, it is right below.

**Comment by thiagokokada** [comment]
@berbiche Done.

**Review by berbiche — APPROVED** [review]
LGTM!

**Pull request merged**